aarch64_extension.cpp

```diff
@@ -275,6 +328,108 @@ class AArch64ArchitectureExtension : public ArchitectureHook {
     return false;
   }
 
+  bool LiftFMOV(cs_insn* instr, LowLevelILFunction& il) {
+    // 00003edc
+    cs_arm64* detail = &(instr->detail->arm64);
+
+    if (detail->op_count != 2) {
+      return false;
+    }
+
+    if (detail->operands[0].type != ARM64_OP_REG
+        || detail->operands[1].type != ARM64_OP_REG) {
+      return false;
+    }
+
+    uint32_t Rd = this->m_base->GetRegisterByName(
+        cs_reg_name(mCapstone, detail->operands[0].reg));
+    uint32_t Rn = this->m_base->GetRegisterByName(
+        cs_reg_name(mCapstone, detail->operands[1].reg));
+
+    size_t Rd_size = this->m_base->GetRegisterInfo(Rd).size;
+    size_t Rn_size = this->m_base->GetRegisterInfo(Rd).size;
+
+    if (Rd_size != Rn_size) {
+      return false;
+    }
+
+    il.AddInstruction(
+        il.SetRegister(Rd_size, Rd, il.Register(Rd_size, Rn)));
+
+    return true;
+  }
+
+  bool LiftMRS(cs_insn* instr, LowLevelILFunction& il) {
+    // 00003edc
+    cs_arm64* detail = &(instr->detail->arm64);
+
+    if (detail->op_count != 2) {
+      return false;
+    }
+
+    if (detail->operands[0].type != ARM64_OP_REG
+        || detail->operands[1].type != ARM64_OP_REG_MRS) {
+      return false;
+    }
+
+    uint32_t Rd = this->m_base->GetRegisterByName(
+        cs_reg_name(mCapstone, detail->operands[0].reg));
+    uint32_t Rn = detail->operands[1].reg;
+
+    size_t Rd_size = this->m_base->GetRegisterInfo(Rd).size;
+    size_t Rn_size = 8;
+
+    if (Rd_size != Rn_size) {
+      return false;
+    }
+
+    il.AddInstruction(
+        il.Intrinsic(vector<RegisterOrFlag> { RegisterOrFlag(false, Rd) },
+                     mIntrinsicNames["__builtin_mrs"],
+                     vector<ExprId> { il.Const(Rn_size, Rn, ILSourceLocation(instr->address, 1)) },
+                     0,
+                     ILSourceLocation(instr->address, 0) ));
+
+    return true;
+  }
+
+  bool LiftMSR(cs_insn* instr, LowLevelILFunction& il) {
+    // 00003edc
+    cs_arm64* detail = &(instr->detail->arm64);
+
+    if (detail->op_count != 2) {
+      return false;
+    }
+
+    if (detail->operands[1].type != ARM64_OP_REG
```

Comment:
This instruction has both a [MSR (immediate)](https://developer.arm.com/docs/100069/0607/a64-general-instructions/msr-immediate) and [MSR (register)](https://developer.arm.com/docs/100069/0607/a64-general-instructions/msr-register) variant. Can we support both?